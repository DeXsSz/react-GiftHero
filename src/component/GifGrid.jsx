import React from 'react'
import GifGridItem from './GifGridItem'
import PropTypes from 'prop-types'
import { useFecthGifs } from '../hooks/useFetchGifs'

const GifGrid = ({ category }) => {
    const { data: images, loading } = useFecthGifs(category);
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {
                loading && <p>Loadingg...</p>
            }
            <div className="cardGrid">

                {
                    images.map(item => {
                        return <GifGridItem key={item.id}
                            {...item}
                        />
                    })
                }
            </div>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
