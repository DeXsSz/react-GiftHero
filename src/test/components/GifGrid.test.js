import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import GifGrid from '../../component/GifGrid';
import { useFecthGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
describe('pruebas en GifGrid.js', () => {
    const category = 'One Punch';

    test('debe mostrar el componente de GifGrid correctamente', () => {
        useFecthGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar items cuando se carguen las img useFecthGifs', () => {
        const gifs = [
            {
                id: 'ABC',
                title: 'Cualquier Cosa',
                url: 'https://localhost/cualquier/cosa.jpg',
            },
        ];
        useFecthGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
