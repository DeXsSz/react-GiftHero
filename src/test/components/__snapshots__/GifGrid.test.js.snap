// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGrid.js debe mostrar el componente de GifGrid correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p>
    Loadingg...
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;
